### PR TITLE
Defer record deletion until the end of hud import

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

**Comment by awisema** [comment]
Needs a test case that has data that is soft deleted and not restored, as the current import tests pass even if I disable `delete_remaining_pending_deletes`?

File: app/models/importers/hmis_six_one_one/base.rb [hunk 1]
```diff
@@ -91,16 +91,35 @@ def import!
...
+          cleanup_any_pending_deletes()
           remove_import_files() if @remove_files
         end
       end # end with_advisory_lock
     end
 
+    def delete_remaining_pending_deletes()
+      # If a pending delete is still present, the associated record is not in the import, and should be
+      # marked as deleted
+      soft_deletable_sources.each do |source|
```
Comment (eanders) [inline app/models/importers/hmis_six_one_one/base.rb]: We should be able to do this completely in sql
`update table_name set DateDeleted = pending_date_deleted where pending_date_deleted is not null`  (in Arel of course)

Comment (eanders) [reply app/models/importers/hmis_six_one_one/base.rb]: `source.where.not(pending_date_deleted: :nil).update_all(DateDeleted: source.arel_table[:pending_date_deleted], pending_date_deleted: nil)` should be pretty close.

File: app/models/importers/hmis_six_one_one/base.rb [hunk 2]
```diff
@@ -465,6 +484,14 @@ def setup_summary(file)
       }
     end
 
+    def soft_deletable_sources
```
Comment (eanders) [inline app/models/importers/hmis_six_one_one/base.rb]: can this just be 
```
def soft_deleteable_sources
  self.class.soft_deleteable_sources
end```
?

**Pull request merged**